### ISSUE TITLE
Enable `pretrained=True` with different num_classes for head.

### ISSUE THREAD

**Issue by tchaton** [facebookresearch/pytorchvideo#26]
## 🚀 Feature

NOTE: Please look at the existing list of Issues tagged with the label ['enhancement`](https://github.com/facebookresearch/pytorchvideo/issues?q=label%3Aenhancement). **Only open a new issue if you do not see your feature request there**.

Currently, the model available don't support `pretrained=True` and `num_classes={something_different_than_orignal_model}` as load_state_dict is using `load_state_dict=True`

## Pitch

NOTE: we only consider adding new features if they are useful for many users.

**Comment by guglielmocamporese** [comment]
Hey,

as a simple workaround, here I reported an example of the code for loading a pretrained `SlowFast` network using a custom number of classes.

Hope this can help!

```python
# Imports
from pytorchvideo.models.slowfast import create_slowfast
import torch.hub

# Custom func for loading a pretrained model, with custom num classes
def make_slowfast(model_name, pretrained=False, **kwargs):
    model = create_slowfast(**kwargs)
    if pretrained:
        model_dict = model.state_dict()
        pretrained_dict = torch.hub.load('facebookresearch/pytorchvideo', 
                                         model_name, 
                                         pretrained=True).state_dict()
        pretrained_dict = {
            k: v for k, v in pretrained_dict.items() 
            if k in list(model_dict.keys()) and model_dict[k].shape == v.shape
        }
        model_dict.update(pretrained_dict) 
        model.load_state_dict(model_dict)
        print('Loaded pretrained model.')
    return model

# Create the model
model = make_slowfast(model_name='slowfast_r50', pretrained=True, 
                      model_num_class=1000)

# Test the model
x_fast = torch.randn(2, 3, 32, 224, 224) # shape [2, 3, 32, 224, 224]
x_slow = x_fast[:, :, ::4] # shape [2, 3, 8, 224, 224]
logits = model([x_slow, x_fast]) # shape [2, 1000]
```
